File: static/js/staff/staff_profile.js
$(document).ready(function() {
    // Initialize all functionality
    init();
    
    // Load profile data on page load
    loadProfileData();
});

// Initialize all functionality
function init() {
    initMobileNavigation();
    initEventListeners();
    initModals(); // Initialize modal functionality
}

// Mobile Navigation Functionality
function initMobileNavigation() {
    const hamburgerMenu = document.getElementById('hamburgerMenu');
    const mobileNav = document.getElementById('mobileNav');
    const closeMobileNav = document.getElementById('closeMobileNav');
    
    if (hamburgerMenu && mobileNav) {
        hamburgerMenu.addEventListener('click', function() {
            mobileNav.classList.add('active');
            document.body.style.overflow = 'hidden';
        });
        
        if (closeMobileNav) {
            closeMobileNav.addEventListener('click', function() {
                mobileNav.classList.remove('active');
                document.body.style.overflow = '';
            });
        }
        
        // Close mobile nav when clicking on links
        const mobileNavLinks = document.querySelectorAll('.mobile-nav-links a');
        mobileNavLinks.forEach(link => {
            link.addEventListener('click', function() {
                mobileNav.classList.remove('active');
                document.body.style.overflow = '';
            });
        });
        
        // Expandable mobile menu sections
        const mobileNavHeaders = document.querySelectorAll('.mobile-nav-header-link');
        mobileNavHeaders.forEach(header => {
            header.addEventListener('click', function() {
                const section = this.getAttribute('data-section');
                const submenu = document.getElementById(`${section}-submenu`);
                const chevron = this.querySelector('.chevron-icon');
                
                // Toggle active class
                this.classList.toggle('active');
                
                // Toggle submenu
                if (submenu) {
                    submenu.classList.toggle('active');
                }
                
                // Rotate only the chevron icon
                if (chevron) {
                    chevron.style.transform = this.classList.contains('active') ? 'rotate(180deg)' : 'rotate(0deg)';
                }
            });
        });
        
        // Close mobile nav when clicking outside
        document.addEventListener('click', function(e) {
            if (!hamburgerMenu.contains(e.target) && !mobileNav.contains(e.target) && mobileNav.classList.contains('active')) {
                mobileNav.classList.remove('active');
                document.body.style.overflow = '';
            }
        });
        
        // Close mobile nav with Escape key
        document.addEventListener('keydown', function(e) {
            if (e.key === 'Escape' && mobileNav.classList.contains('active')) {
                mobileNav.classList.remove('active');
                document.body.style.overflow = '';
            }
        });
    }
}

// Initialize modal functionality
function initModals() {
    // Close modal function - works for ALL modals
    function closeAllModals() {
        $('.modal').fadeOut(300);
        document.body.style.overflow = '';
        document.body.classList.remove('modal-open');
    }

    // Open modal function
    function openModal(modalId) {
        const modal = document.getElementById(modalId);
        if (modal) {
            modal.style.display = 'flex';
            document.body.style.overflow = 'hidden';
            document.body.classList.add('modal-open');
        }
    }

    // Close modal when clicking outside - works for ALL modals
    $(document).on('click', function(e) {
        if ($(e.target).hasClass('modal')) {
            closeAllModals();
        }
    });

    // Escape key to close modals - works for ALL modals
    $(document).keyup(function(e) {
        if (e.keyCode === 27) {
            closeAllModals();
        }
    });

    // ===== SPECIFIC MODAL FUNCTIONALITY =====

    // Logout Modal
    $('#logout-trigger').click(function(e) {
        e.preventDefault();
        e.stopPropagation();
        openModal('logout-modal');
    });
    
    $('#mobile-logout-trigger').click(function(e) {
        e.preventDefault();
        e.stopPropagation();
        // First close mobile nav properly
        const mobileNav = document.getElementById('mobileNav');
        if (mobileNav) {
            mobileNav.classList.remove('active');
        }
        // Then open logout modal
        setTimeout(() => {
            openModal('logout-modal');
        }, 10);
    });
    
    $('#cancel-logout').click(function(e) {
        e.preventDefault();
        e.stopPropagation();
        closeAllModals();
    });
    
    $('#close-logout-modal').click(function(e) {
        e.preventDefault();
        e.stopPropagation();
        closeAllModals();
    });
    
    $('#confirm-logout').click(function(e) {
        e.preventDefault();
        e.stopPropagation();
        const logoutUrl = document.body.getAttribute('data-logout-url');
        if (logoutUrl) {
            window.location.href = logoutUrl;
        } else {
            console.error('Logout URL not found');
            window.location.href = "/logout";
        }
    });
}

// Initialize event listeners
function initEventListeners() {
    // Profile picture handling
    $('#profile_picture').change(function() {
        const file = this.files[0];
        if (file) {
            // Check file size (max 10MB)
            if (file.size > 10 * 1024 * 1024) {
                showMessage('error', 'File size exceeds 10MB limit');
                $(this).val('');
                return;
            }
            
            // Check file type
            const validTypes = ['image/jpeg', 'image/png', 'image/jpg'];
            if (!validTypes.includes(file.type)) {
                showMessage('error', 'Only JPG, JPEG, or PNG files are allowed');
                $(this).val('');
                return;
            }
            
            const reader = new FileReader();
            reader.onload = function(e) {
                $('#profile-preview').attr('src', e.target.result);
            }
            reader.readAsDataURL(file);
        }
    });
    
    // Remove profile picture
    $('#remove-picture').click(function() {
        $('#profile_picture').val('');
        $('#profile-preview').attr('src', defaultProfilePic);
        showMessage('info', 'Profile picture removed. Remember to save changes.');
    });
    
    // Signature handling - UPDATED with status awareness
    $('#signature').change(function () {
        const file = this.files[0];
        if (!file) return;

        // Check file size (max 10MB)
        if (file.size > 10 * 1024 * 1024) {
            showMessage('error', 'File size exceeds 10MB limit');
            this.value = '';
            return;
        }

        // Check file type
        if (!['image/png','image/jpeg','image/jpg'].includes(file.type)) {
            showMessage('error', 'Signature must be PNG or JPG');
            this.value = '';
            return;
        }

        // Show warning about status reset if signature already exists
        const currentStatus = $('#signatureStatusBadge').data('status');
        if (currentStatus && currentStatus !== 'none') {
            Swal.fire({
                title: 'Reset Verification Status?',
                text: 'Uploading a new signature will reset your verification status to "Pending". You will need to wait for admin approval again.',
                icon: 'warning',
                showCancelButton: true,
                confirmButtonColor: '#003366',
                cancelButtonColor: '#6c757d',
                confirmButtonText: 'Yes, upload anyway',
                cancelButtonText: 'Cancel'
            }).then((result) => {
                if (result.isConfirmed) {
                    proceedWithSignatureUpload(file);
                } else {
                    $('#signature').val('');
                }
            });
        } else {
            proceedWithSignatureUpload(file);
        }
    });
    
    // Remove signature
    $('#remove-signature').click(function() {
        const currentStatus = $('#signatureStatusBadge').data('status');
        
        if (currentStatus && currentStatus !== 'none') {
            Swal.fire({
                title: 'Remove Signature?',
                text: 'Removing your signature will require you to upload a new one and go through verification again.',
                icon: 'warning',
                showCancelButton: true,
                confirmButtonColor: '#b91c1c',
                cancelButtonColor: '#6c757d',
                confirmButtonText: 'Yes, remove',
                cancelButtonText: 'Cancel'
            }).then((result) => {
                if (result.isConfirmed) {
                    $('#signature').val('');
                    $('#signature-preview').attr('src', defaultSignaturePic);
                    updateSignatureStatus('none', 'No signature uploaded', 
                        'You haven\'t uploaded a signature yet. Please upload your e-signature for document verification.');
                    showMessage('info', 'Signature removed. Remember to save changes.');
                }
            });
        } else {
            $('#signature').val('');
            $('#signature-preview').attr('src', defaultSignaturePic);
            updateSignatureStatus('none', 'No signature uploaded', 
                'You haven\'t uploaded a signature yet. Please upload your e-signature for document verification.');
            showMessage('info', 'Signature removed. Remember to save changes.');
        }
    });
    
    // Show/hide password
    $('.toggle-password').click(function() {
        const input = $(this).siblings('input');
        const type = input.attr('type') === 'password' ? 'text' : 'password';
        input.attr('type', type);
        $(this).toggleClass('fa-eye fa-eye-slash');
    });
    
    // Password strength indicator
    $('#new_password').on('input', function() {
        const password = $(this).val();
        let strength = 0;
        let text = '';
        
        if (password.length > 0) {
            if (password.length >= 8) strength++;
            if (/[A-Z]/.test(password)) strength++;
            if (/[0-9]/.test(password)) strength++;
            if (/[^A-Za-z0-9]/.test(password)) strength++;
            
            const strengthText = ['Very Weak', 'Weak', 'Medium', 'Strong', 'Very Strong'];
            text = strengthText[strength];
            
            $('.strength-meter .strength-bar').removeClass('active');
            for (let i = 0; i < strength; i++) {
                $('.strength-bar').eq(i).addClass('active');
            }
            
            $('.strength-text span').text(text);
        } else {
            $('.strength-meter .strength-bar').removeClass('active');
            $('.strength-text span').text('None');
        }
    });
    
    // Form submission
    $('#profile-form').on('submit', function(e) {
        e.preventDefault();
        
        // Check password match
        const newPassword = $('#new_password').val();
        const confirmPassword = $('#confirm_password').val();
        
        if (newPassword && newPassword !== confirmPassword) {
            showMessage('error', 'New password and confirmation do not match');
            return;
        }
        
        // Show loading state
        $(this).addClass('loading');
        $('.save-btn').html('<i class="fas fa-spinner fa-spin"></i> Saving...');
        
        let formData = new FormData(this);
        
        $.ajax({
            url: updateProfileUrl,
            type: 'POST',
            data: formData,
            processData: false,
            contentType: false,
            success: function(response) {
                if (response.success) {
                    showMessage('success', response.message);
                    if (response.updated_profile) {
                        updateFormFields(response.updated_profile);
                        
                        // Show additional message if signature was updated
                        if (response.signature_updated) {
                            setTimeout(() => {
                                Swal.fire({
                                    icon: 'info',
                                    title: 'Signature Updated',
                                    text: 'Your signature has been uploaded and is now pending admin verification.',
                                    confirmButtonColor: '#003366'
                                });
                            }, 500);
                        }
                    }
                } else {
                    showMessage('error', response.error || 'Update failed');
                }
                $('#profile-form').removeClass('loading');
                $('.save-btn').html('<i class="fas fa-save"></i> Save Changes');
            },
            error: function(xhr) {
                let errorMsg = 'An error occurred while updating your profile';
                if (xhr.responseJSON && xhr.responseJSON.error) {
                    errorMsg = xhr.responseJSON.error;
                } else if (xhr.statusText) {
                    errorMsg = xhr.statusText;
                }
                showMessage('error', errorMsg);
                $('#profile-form').removeClass('loading');
                $('.save-btn').html('<i class="fas fa-save"></i> Save Changes');
            }
        });
    });
}

// Helper function to proceed with signature upload
function proceedWithSignatureUpload(file) {
    const reader = new FileReader();
    reader.onload = function(e) {
        $('#signature-preview').attr('src', e.target.result);
        // Update status to pending temporarily (will be confirmed on save)
        updateSignatureStatus('pending', 'Pending Verification', 
            'Your signature will be pending verification after you save changes.');
    };
    reader.readAsDataURL(file);
}

// Load profile data on page load
function loadProfileData() {
    // Show loading state
    $('#profile-form').addClass('loading');
    
    $.ajax({
        url: fetchProfileUrl,
        type: 'GET',
        success: function(response) {
            if (response.success && response.staff) {
                updateFormFields(response.staff);
            } else {
                showMessage('error', response.error || 'Failed to load profile data');
            }
            $('#profile-form').removeClass('loading');
        },
        error: function(xhr) {
            showMessage('error', 'Failed to load profile data. Please try again later.');
            $('#profile-form').removeClass('loading');
        }
    });
}

// Update form fields with profile data
function updateFormFields(profileData) {
    $('#username').val(profileData.username || '');
    $('#email').val(profileData.email || '');
    $('#first_name').val(profileData.first_name || '');
    $('#middle_name').val(profileData.middle_name || '');
    $('#last_name').val(profileData.last_name || '');
    $('#contact_number').val(profileData.contact_number || '');
    $('#province').val(profileData.province || '');
    $('#municipality').val(profileData.municipality || '');
    $('#baranggay').val(profileData.baranggay || '');
    $('#date_of_birth').val(profileData.date_of_birth || '');
    $('#gender').val(profileData.gender || '');
    
    // Update profile picture
    if (profileData.profile_picture_url) {
        $('#profile-preview').attr('src', profileData.profile_picture_url);
        $('#nav-profile-pic').attr('src', profileData.profile_picture_url);
    } else {
        $('#profile-preview').attr('src', defaultProfilePic);
        $('#nav-profile-pic').attr('src', defaultProfilePic);
    }
    
    // Update signature with status
    if (profileData.signature_url) {
        $('#signature-preview').attr('src', profileData.signature_url);
    } else {
        $('#signature-preview').attr('src', defaultSignaturePic);
    }
    
    // Update signature status display
    if (profileData.signature_status) {
        updateSignatureStatus(
            profileData.signature_status,
            profileData.signature_status_text,
            profileData.signature_status_description
        );
    }
}

// Update signature status UI
function updateSignatureStatus(status, statusText, description) {
    const badge = $('#signatureStatusBadge');
    const frame = $('#signatureFrame');
    const descriptionDiv = $('#signatureStatusDescription');
    const warningDiv = $('#signatureWarning');
    const warningMessage = $('#warningMessage');
    const uploadBtn = $('#signatureUploadBtn');
    
    // Remove all status classes
    badge.removeClass('status-none status-pending status-verified status-rejected');
    frame.removeClass('status-none status-pending status-verified status-rejected');
    
    // Add appropriate class
    badge.addClass(`status-${status}`);
    frame.addClass(`status-${status}`);
    
    // Update badge text and icon
    badge.find('.status-text').text(statusText);
    
    // Store status in data attribute
    badge.data('status', status);
    
    // Show badge
    badge.show();
    
    // Update description
    descriptionDiv.text(description).show();
    
    // Handle warnings and upload button state based on status
    if (status === 'pending') {
        warningMessage.text('Your signature is pending approval. Uploading a new signature will reset this status.');
        warningDiv.show();
        uploadBtn.removeClass('disabled');
    } else if (status === 'verified') {
        warningMessage.text('Your signature is verified. Uploading a new signature will reset it to pending status.');
        warningDiv.show();
        uploadBtn.removeClass('disabled');
    } else if (status === 'rejected') {
        warningMessage.text('Your signature was rejected. Please upload a new signature following the guidelines.');
        warningDiv.show();
        uploadBtn.removeClass('disabled');
    } else {
        warningDiv.hide();
        uploadBtn.removeClass('disabled');
    }
}

// SweetAlert2 Message Handling
function showMessage(type, text) {
    const swalConfig = {
        title: getTitleByType(type),
        text: text,
        icon: type,
        confirmButtonText: 'OK',
        confirmButtonColor: getButtonColorByType(type),
        customClass: {
            popup: 'sweetalert-popup',
            title: 'sweetalert-title',
            confirmButton: 'sweetalert-confirm-btn'
        }
    };
    
    Swal.fire(swalConfig);
}

function getTitleByType(type) {
    switch(type) {
        case 'success': return 'Success';
        case 'error': return 'Error';
        case 'info': return 'Information';
        default: return 'Message';
    }
}

function getButtonColorByType(type) {
    switch(type) {
        case 'success': return '#15803d';
        case 'error': return '#b91c1c';
        case 'info': return '#1d4ed8';
        default: return '#003366';
    }
}

File: staff_profile.py
from flask import Blueprint, request, jsonify, session, current_app, url_for, render_template
from database import get_db
import os
from werkzeug.utils import secure_filename
import uuid
from datetime import datetime
import traceback

staff_profile_bp = Blueprint('staff_profile', __name__, url_prefix='/staff')

ALLOWED_EXTENSIONS = {'png', 'jpg', 'jpeg', 'gif'}
ALLOWED_SIGNATURE_EXTENSIONS = {'png', 'jpg', 'jpeg'}

def allowed_file(filename):
    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS

def allowed_signature(filename):
    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_SIGNATURE_EXTENSIONS
 
@staff_profile_bp.route('/profile', methods=['GET'])
def staff_profile_page():
    """Render the HTML profile page"""
    if 'user_id' not in session or session.get('role') != 'staff': 
        return render_template('login.html', error='Unauthorized access')
     
    return render_template('staffs/staff_profile.html')
 
@staff_profile_bp.route('/profile/data', methods=['GET'])
def staff_profile_data():
    """Return JSON data for the profile (used by AJAX)"""
    if 'user_id' not in session or session.get('role') != 'staff':
        return jsonify({'error': 'Unauthorized access'}), 401

    user_id = session['user_id']
    db = get_db()
    cursor = db.cursor(dictionary=True)

    try:
        query = """
            SELECT l.user_id, l.username, l.email, l.role, l.account_status, l.verified,
                   pi.info_id, pi.first_name, pi.middle_name, pi.last_name,
                   pi.province, pi.municipality, pi.baranggay,
                   pi.contact_number, pi.date_of_birth, pi.gender,
                   pi.profile_picture, pi.signature, pi.date_registered
            FROM login l
            JOIN personal_information pi ON l.user_id = pi.user_id
            WHERE l.user_id = %s
        """
        cursor.execute(query, (user_id,))
        staff = cursor.fetchone()

        if not staff:
            return jsonify({'error': 'Staff profile not found'}), 404

        staff['date_of_birth'] = staff['date_of_birth'].strftime('%Y-%m-%d') if staff['date_of_birth'] else None
        staff['date_registered'] = staff['date_registered'].strftime('%Y-%m-%d %H:%M:%S') if staff['date_registered'] else None

        staff['profile_picture_url'] = url_for(
            'static',
            filename=f"uploads/profile_pictures/{staff['profile_picture']}",
            _external=True
        ) if staff['profile_picture'] else None

        staff['signature_url'] = url_for(
            'static',
            filename=f"uploads/signatures/{staff['signature']}",
            _external=True
        ) if staff['signature'] else None
        
        # Determine signature status
        if not staff['signature']:
            staff['signature_status'] = 'none'
            staff['signature_status_text'] = 'No signature uploaded'
            staff['signature_status_description'] = 'You haven\'t uploaded a signature yet. Please upload your e-signature for document verification.'
        elif staff['verified'] == 'verified':
            staff['signature_status'] = 'verified'
            staff['signature_status_text'] = 'Verified'
            staff['signature_status_description'] = '✓ Your signature has been verified and approved for official documents.'
        elif staff['verified'] == 'rejected':
            staff['signature_status'] = 'rejected'
            staff['signature_status_text'] = 'Rejected'
            staff['signature_status_description'] = '✗ Your signature was rejected. Please upload a new signature following the guidelines.'
        else:  # pending or null
            staff['signature_status'] = 'pending'
            staff['signature_status_text'] = 'Pending Verification'
            staff['signature_status_description'] = 'Your signature is pending admin verification. You\'ll be notified once approved.'

        return jsonify({'success': True, 'staff': staff})

    except Exception as e:
        current_app.logger.error(traceback.format_exc())
        return jsonify({'error': 'Failed to fetch profile'}), 500
    finally:
        cursor.close()


@staff_profile_bp.route('/profile/update', methods=['POST'])
def update_staff_profile():
    if 'user_id' not in session or session.get('role') != 'staff':
        return jsonify({'error': 'Unauthorized access'}), 401

    user_id = session['user_id']
    db = get_db()
    cursor = db.cursor(dictionary=True)

    try:
        data = request.form.to_dict()
        files = request.files

        cursor.execute("SELECT password, username, email FROM login WHERE user_id=%s", (user_id,))
        user = cursor.fetchone()
        if not user:
            return jsonify({'error': 'User not found'}), 404
 
        # Handle profile picture upload (overwrite)
        profile_picture = None
        if 'profile_picture' in files:
            file = files['profile_picture']
            if file and file.filename and allowed_file(file.filename):
                ext = file.filename.rsplit('.', 1)[1].lower()
                filename = f"{user_id}_{uuid.uuid4().hex}.{ext}"
                upload_dir = os.path.join(current_app.static_folder, 'uploads', 'profile_pictures')
                os.makedirs(upload_dir, exist_ok=True)
                
                # Delete old profile picture if exists
                cursor.execute("SELECT profile_picture FROM personal_information WHERE user_id=%s", (user_id,))
                old = cursor.fetchone()
                if old and old['profile_picture']:
                    old_path = os.path.join(upload_dir, old['profile_picture'])
                    if os.path.exists(old_path):
                        os.remove(old_path)
                
                file.save(os.path.join(upload_dir, secure_filename(filename)))
                profile_picture = filename
 
        # Handle signature upload (overwrite) - this will reset verification status
        signature_file = None
        signature_updated = False
        if 'signature' in files:
            file = files['signature']
            if file and file.filename and allowed_signature(file.filename):
                ext = file.filename.rsplit('.', 1)[1].lower()
                filename = f"{user_id}_{uuid.uuid4().hex}.{ext}"
                upload_dir = os.path.join(current_app.static_folder, 'uploads', 'signatures')
                os.makedirs(upload_dir, exist_ok=True)
                
                # Delete old signature if exists
                cursor.execute("SELECT signature FROM personal_information WHERE user_id=%s", (user_id,))
                old = cursor.fetchone()
                if old and old['signature']:
                    old_path = os.path.join(upload_dir, old['signature'])
                    if os.path.exists(old_path):
                        os.remove(old_path)
                
                file.save(os.path.join(upload_dir, secure_filename(filename)))
                signature_file = filename
                signature_updated = True
 
        # Handle password update
        password_update = ""
        password_params = ()
        if data.get('current_password') and data.get('new_password'):
            if data['current_password'] != user['password']:
                return jsonify({'error': 'Incorrect current password'}), 400
            password_update = ", password=%s"
            password_params = (data['new_password'],)

        # Update login table
        cursor.execute(
            f"UPDATE login SET username=%s, email=%s {password_update} WHERE user_id=%s",
            (data.get('username', user['username']),
             data.get('email', user['email'])) + password_params + (user_id,)
        )

        # If signature was updated, reset verification status to pending
        if signature_updated:
            cursor.execute("""
                UPDATE login SET verified='pending' 
                WHERE user_id=%s AND role='staff'
            """, (user_id,))

        # Prepare fields for personal_information update
        fields = {
            'first_name': data.get('first_name', ''),
            'middle_name': data.get('middle_name', ''),
            'last_name': data.get('last_name', ''),
            'contact_number': data.get('contact_number', ''),
            'province': data.get('province', ''),
            'municipality': data.get('municipality', ''),
            'baranggay': data.get('baranggay', ''),
            'gender': data.get('gender', ''),
            'date_of_birth': datetime.strptime(data['date_of_birth'], '%Y-%m-%d').date() if data.get('date_of_birth') else None,
            'user_id': user_id
        }

        # Build dynamic SQL for personal_information update
        update_fields = []
        update_values = []
        
        for key, value in fields.items():
            if key != 'user_id':
                update_fields.append(f"{key}=%s")
                update_values.append(value)
        
        if profile_picture:
            update_fields.append("profile_picture=%s")
            update_values.append(profile_picture)
            
        if signature_file:
            update_fields.append("signature=%s")
            update_values.append(signature_file)
        
        update_values.append(user_id)
        
        if update_fields:
            cursor.execute(f"""
                UPDATE personal_information SET
                    {', '.join(update_fields)}
                WHERE user_id=%s
            """, update_values)

        db.commit()
         
        # Fetch updated profile
        query = """
            SELECT l.username, l.email, l.role, l.account_status, l.verified,
                   pi.first_name, pi.middle_name, pi.last_name,
                   pi.province, pi.municipality, pi.baranggay,
                   pi.contact_number, pi.date_of_birth, pi.gender,
                   pi.profile_picture, pi.signature
            FROM login l
            JOIN personal_information pi ON l.user_id = pi.user_id
            WHERE l.user_id = %s
        """
        cursor.execute(query, (user_id,))
        updated_profile = cursor.fetchone()
        
        if updated_profile and updated_profile['date_of_birth']:
            updated_profile['date_of_birth'] = updated_profile['date_of_birth'].strftime('%Y-%m-%d')
        
        updated_profile['profile_picture_url'] = url_for(
            'static',
            filename=f"uploads/profile_pictures/{updated_profile['profile_picture']}",
            _external=True
        ) if updated_profile['profile_picture'] else None
        
        updated_profile['signature_url'] = url_for(
            'static',
            filename=f"uploads/signatures/{updated_profile['signature']}",
            _external=True
        ) if updated_profile['signature'] else None
        
        # Determine signature status for updated profile
        if not updated_profile['signature']:
            updated_profile['signature_status'] = 'none'
            updated_profile['signature_status_text'] = 'No signature uploaded'
            updated_profile['signature_status_description'] = 'You haven\'t uploaded a signature yet. Please upload your e-signature for document verification.'
        elif updated_profile['verified'] == 'verified':
            updated_profile['signature_status'] = 'verified'
            updated_profile['signature_status_text'] = 'Verified'
            updated_profile['signature_status_description'] = '✓ Your signature has been verified and approved for official documents.'
        elif updated_profile['verified'] == 'rejected':
            updated_profile['signature_status'] = 'rejected'
            updated_profile['signature_status_text'] = 'Rejected'
            updated_profile['signature_status_description'] = '✗ Your signature was rejected. Please upload a new signature following the guidelines.'
        else:  # pending or null
            updated_profile['signature_status'] = 'pending'
            updated_profile['signature_status_text'] = 'Pending Verification'
            updated_profile['signature_status_description'] = 'Your signature is pending admin verification. You\'ll be notified once approved.'

        return jsonify({
            'success': True, 
            'message': 'Profile updated successfully',
            'updated_profile': updated_profile,
            'signature_updated': signature_updated,
            'new_status': 'pending' if signature_updated else None
        })

    except Exception as e:
        db.rollback()
        current_app.logger.error(traceback.format_exc())
        return jsonify({'error': 'Update failed'}), 500
    finally:
        cursor.close()